subcommand/server-acl-init/command_test.go

```diff
@@ -721,318 +638,100 @@ func TestRun_BindingRuleUpdates(t *testing.T) {
-				},
-			},
-		},
+	// Check that it has the right policies.
+	consul, err := api.NewClient(&api.Config{
+		Address: srv.HTTPAddr,
 	})
 	require.NoError(err)
-	_, err = k8s.AppsV1().StatefulSets(ns).Create(&appv1.StatefulSet{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: resourcePrefix + "-server",
-			Labels: map[string]string{
-				"component": "server",
-				"app":       "consul",
-				"release":   releaseName,
-			},
-		},
-		Status: appv1.StatefulSetStatus{
-			UpdateRevision:  "updated",
-			CurrentRevision: "current",
-		},
-	})
+	tokenData, _, err := consul.ACL().TokenReadSelf(&api.QueryOptions{Token: bootToken})
 	require.NoError(err)
+	require.Equal("global-management", tokenData.Policies[0].Name)
 
-	ui := cli.NewMockUi()
-	cmd := Command{
-		UI:        ui,
-		clientset: k8s,
-	}
-	cmd.init()
-
-	// Start the command before the Pod exist.
-	// Run in a goroutine so we can create the Pods asynchronously
-	done := make(chan bool)
-	var responseCode int
-	go func() {
-		responseCode = cmd.Run([]string{
-			"-server-label-selector=component=server,app=consul,release=" + releaseName,
-			"-resource-prefix=" + resourcePrefix,
-			"-k8s-namespace=" + ns,
-			"-expected-replicas=1",
-		})
-		close(done)
-	}()
-
-	// Asynchronously update the deployment status after a delay.
-	go func() {
-		// Update after a delay between 100 and 500ms.
-		// It's randomized to ensure we're not relying on specific timing.
-		delay := 100 + rand.Intn(400)
-		time.Sleep(time.Duration(delay) * time.Millisecond)
-		_, err = k8s.AppsV1().StatefulSets(ns).Update(&appv1.StatefulSet{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: resourcePrefix + "-server",
-				Labels: map[string]string{
-					"component": "server",
-					"app":       "consul",
-					"release":   releaseName,
-				},
-			},
-			Status: appv1.StatefulSetStatus{
-				UpdateRevision:  "updated",
-				CurrentRevision: "updated",
-			},
-		})
-		require.NoError(err)
-	}()
-
-	// Wait for the command to exit.
-	select {
-	case <-done:
-		require.Equal(0, responseCode, ui.ErrorWriter.String())
-	case <-time.After(2 * time.Second):
-		require.FailNow("command did not exit after 2s")
+	// Check that the agent policy was created.
+	policies, _, err := consul.ACL().PolicyList(&api.QueryOptions{Token: bootToken})
+	require.NoError(err)
+	found := false
+	for _, p := range policies {
+		if p.Name == "agent-token" {
+			found = true
+			break
+		}
 	}
-
-	// Test that the bootstrap kube secret is created.
-	getBootToken(t, k8s, resourcePrefix, ns)
-
-	// Test that the expected API calls were made.
-	require.Equal([]APICall{
-		{
-			"PUT",
-			"/v1/acl/bootstrap",
-		},
-		{
-			"PUT",
-			"/v1/acl/policy",
-		},
-		{
-			"PUT",
-			"/v1/acl/token",
-		},
-		{
-			"PUT",
-			"/v1/agent/token/agent",
-		},
-		{
-			"GET",
-			"/v1/agent/self",
-		},
-		{
-			"PUT",
-			"/v1/acl/policy",
-		},
-		{
-			"PUT",
-			"/v1/acl/token",
-		},
-	}, consulAPICalls)
+	require.True(found, "agent-token policy was not found")
```

Comment:
💯 wow quite the refactor. The one thing we're not testing anymore is that the servers get the ACL token applied to themselves (the `PUT /v1/agent/token/agent` assertion above). That's why I used an http server here instead of the consul test server.

This is still tested like that in the noleader and clienttokensretry tests so we're okay to not have it here.